server/lib/parties/party-service.test.ts

```diff
@@ -0,0 +1,264 @@
+import { NydusServer } from 'nydus'
+import { RequestSessionLookup } from '../websockets/session-lookup'
+import { ClientSocketsManager, UserSocketsManager } from '../websockets/socket-groups'
+import {
+  clearTestLogs,
+  createFakeNydusServer,
+  InspectableNydusClient,
+  NydusConnector,
+} from '../websockets/testing/websockets'
+import PartyService, { getInvitesPath, getPartyPath, PartyRecord, PartyUser } from './party-service'
+
+describe('parties/party-service', () => {
+  const user1: PartyUser = { id: 1, name: 'pachi' }
+  const user2: PartyUser = { id: 2, name: 'harem' }
+  const user3: PartyUser = { id: 3, name: 'intrigue' }
+  const user4: PartyUser = { id: 4, name: 'tec27' }
+  const user5: PartyUser = { id: 5, name: 'heyoka' }
+  const user6: PartyUser = { id: 6, name: 'hot_bid' }
+  const user7: PartyUser = { id: 7, name: 'royo' }
+  const user8: PartyUser = { id: 8, name: 'riptide' }
+  const user9: PartyUser = { id: 9, name: 'manifesto7' }
+
+  const USER1_CLIENT_ID = 'USER1_CLIENT_ID'
+  const USER2_CLIENT_ID = 'USER2_CLIENT_ID'
+  const USER3_CLIENT_ID = 'USER3_CLIENT_ID'
+  const USER4_CLIENT_ID = 'USER4_CLIENT_ID'
+  const USER5_CLIENT_ID = 'USER5_CLIENT_ID'
+  const USER6_CLIENT_ID = 'USER6_CLIENT_ID'
+  const USER7_CLIENT_ID = 'USER7_CLIENT_ID'
+  const USER8_CLIENT_ID = 'USER8_CLIENT_ID'
+  const USER9_CLIENT_ID = 'USER9_CLIENT_ID'
+
+  let client1: InspectableNydusClient
+  let client2: InspectableNydusClient
+  let client3: InspectableNydusClient
+  let client4: InspectableNydusClient
+  let client5: InspectableNydusClient
+  let client6: InspectableNydusClient
+  let client7: InspectableNydusClient
+  let client8: InspectableNydusClient
+  let client9: InspectableNydusClient
+
+  let nydus: NydusServer
+  let partyService: PartyService
+  let connector: NydusConnector
+
+  beforeEach(() => {
+    nydus = createFakeNydusServer()
+    const sessionLookup = new RequestSessionLookup()
+    const userSocketsManager = new UserSocketsManager(nydus, sessionLookup, async () => {})
+    const clientSocketsManager = new ClientSocketsManager(nydus, sessionLookup)
+    partyService = new PartyService(nydus, clientSocketsManager, userSocketsManager)
+    connector = new NydusConnector(nydus, sessionLookup)
+
+    client1 = connector.connectClient(user1, USER1_CLIENT_ID)
```

Comment:
Wonder if it might be cleaner to just make an array of these things and reference them that way :)

Reply:
Eh, I thought it would be cleaner too, but after having done it I realized it wasn't really :d. Immediate problem I noticed was that now clients are referenced from 0, instead of 1, due to zero-based indexing nature of arrays. This makes it confusing to read when using, for example, `user1` and `client0` in the same test. Then I tried renaming all users to be 0-based as well, but that makes it more annoying when you have to test how many users there are in a party since you have to add 1 to the user index to know how many of them are there. And _then_ I tried adding a dummy client to the array at the index 0, so all other clients could be easily linked to their 1-based users counterpart, and as I was writing a comment explaining why this dummy user is in the array, I realized that this isn't really that cleaner than what it was before :d